packages/firestore/test/lite/integration.test.ts

```diff
@@ -2047,11 +2047,12 @@ describe('withConverter() support', () => {
 
 describe('countQuery()', () => {
   const testDocs = [
-    { author: 'authorA', title: 'titleA' },
-    { author: 'authorA', title: 'titleB' },
-    { author: 'authorB', title: 'titleC' },
-    { author: 'authorB', title: 'titleD' },
-    { author: 'authorB', title: 'titleE' }
+    { id: 3, author: 'authorA', title: 'titleA' },
```

Comment:
I think it's time to get rid of this `testDocs` variable and re-define it in each test case.

Here is the reason: the structure of the documents in this variable have been carefully crafted to suit several different test cases. Minor changes can break one or more test cases inadvertently.

Unlike "normal" code, test code has some different rules. One different rule is that of duplication. In normal code the DRY principle usually applies: "don't repeat yourself". Duplicated logic is best to be factored out into helpers method or shared variables. The same principle does _not_, however, apply for testing code. Test code must be optimized for readability, and code duplication is acceptable when it improves readability. When you have to scroll up a few pages to find the definition of `testDocs` to understand the test, this reduces readability. In this case, it is better to repeat `testDocs` in each test method so that the data is immediately available to the reader. It also localizes any special tweaking to only the relevant test cases.

I like to think of it this way: the correctness of your production code is verified by the tests, but the correctness of the tests is verified by humans via code inspection. Therefore, test code needs to be optimized for human readability to maximize the probability of it being correct.

Take a look at go/unit-test-readability-legacy if you have some time. Not everything in that presentation is relevant, but the guiding principles definitely are.